x-pack/legacy/plugins/reporting/export_types/png/server/lib/generate_png.ts

```diff
@@ -6,45 +6,60 @@
 
 import * as Rx from 'rxjs';
 import { toArray, mergeMap } from 'rxjs/operators';
+import { KbnServer } from '../../../../types';
+// @ts-ignore
 import { oncePerServer } from '../../../../server/lib/once_per_server';
 import { screenshotsObservableFactory } from '../../../common/lib/screenshots';
 import { PreserveLayout } from '../../../common/layouts/preserve_layout';
 
-function generatePngObservableFn(server) {
-  const screenshotsObservable = screenshotsObservableFactory(server);
-  const captureConcurrency = 1;
-
-  const urlScreenshotsObservable = (url, conditionalHeaders, layout, browserTimezone) => {
-    return Rx.of(url).pipe(
-      mergeMap(url => screenshotsObservable({ url, conditionalHeaders, layout, browserTimezone }),
-        (outer, inner) => inner,
-        captureConcurrency
-      )
-    );
-  };
+interface ScreenshotData {
+  base64EncodedData: string;
+}
 
-  const createPngWithScreenshots = async ({ urlScreenshots }) => {
+interface UrlScreenshot {
+  screenshots: ScreenshotData[];
+}
 
+function generatePngObservableFn(server: KbnServer) {
+  const screenshotsObservable = screenshotsObservableFactory(server);
+  const captureConcurrency = 1;
+  const createPngWithScreenshots = async ({
+    urlScreenshots,
+  }: {
+    urlScreenshots: UrlScreenshot[];
+  }) => {
     if (urlScreenshots.length !== 1) {
-      throw new Error(`Expected there to be 1 URL screenshot, but there are ${urlScreenshots.length}`);
+      throw new Error(
+        `Expected there to be 1 URL screenshot, but there are ${urlScreenshots.length}`
+      );
     }
     if (urlScreenshots[0].screenshots.length !== 1) {
-      throw new Error(`Expected there to be 1 screenshot, but there are ${urlScreenshots[0].screenshots.length}`);
+      throw new Error(
+        `Expected there to be 1 screenshot, but there are ${urlScreenshots[0].screenshots.length}`
```

Comment:
maybe we should localize these strings since i assume they are user facing?

Reply:
https://github.com/elastic/kibana/issues/42873